src/Argu/Parsers/Cli.fs

```diff
@@ -121,13 +122,27 @@ type CliParseResultAggregator internal (argInfo : UnionArgInfo, stack : CliParse
             if stack.TryDispatchResult result then ()
             else unrecognizedParseResults.Add result.Value
 
+    member x.AppendResultWithNestedResults caseInfo context arguments nestedResults =
+        missingMandatoryCasesOfNestedResults.AddRange(nestedResults.MissingMandatoryCases)
+        x.AppendResult caseInfo context arguments
+
     member _.AppendUnrecognized(token:string) = unrecognized.Add token
 
     member x.ToUnionParseResults() =
         { Cases = results.Value |> Array.map (fun c -> c.ToArray()) ;
           UnrecognizedCliParams = Seq.toList unrecognized ;
           UnrecognizedCliParseResults = Seq.toList unrecognizedParseResults ;
-          IsUsageRequested = x.IsUsageRequested }
+          IsUsageRequested = x.IsUsageRequested
+          MissingMandatoryCases =
+                argInfo.Cases.Value
+                |> Seq.choose (fun case ->
+                    if case.IsMandatory.Value && results.Value[case.Tag].Count = 0
+                    then Some case
```

Comment:
if returning same thing, Seq.filter
also this whole append - tolist chain is probably best expressed as a list comprehension i.e. approx
```
[ for x in argInfo.Cases do
    if ...
        then x
  yeld! missingMandatory... ]           
  ```

Reply:
done